### PR TITLE
Add integer parsing tests. Fixes #682

### LINKED ISSUE #682: Improve test for backend string parsing functions.
In ga4gh/backend.py, the ``_parseIntegerArgument`` function is responsible for parsing integer values passed as arguments in GET requests.

Add some tests for this in ``tests/unit/test_backends.py``. Add a  test method to the TestPrivateBackendMethods class to test positive results (i.e., do we correctly return the right value?) and another to test negative results (i.e., do we correctly raise a BadRequestIntegerException when called with non-integer values?).

We also want to improve the tests for the _parsePageToken function. We want to include tests for all negative results, as well as improved tests for the positive results. (The existing test is good, but we'd like to check more possibilities, like varying numbers of colon separated values)

### PR DESCRIPTION
**Pull request by andrewyatz**
Ronseal

### CODE REVIEW

**Comment by dcolligan** [comment]
+1

File: tests/unit/test_backends.py [hunk 1]
```diff
@@ -242,3 +242,14 @@ def testParsePageToken(self):
         goodPageToken = "12:34:567:8:9000"
         parsedToken = backend._parsePageToken(goodPageToken, 5)
         self.assertEqual(parsedToken[2], 567)
+
+    def testParseIntegerArgument(self):
+        args = {"good": "1", "negative": "-1", "bad": "A", "badfloat": "0.98"}
```
Comment (jeromekelleher) [inline tests/unit/test_backends.py]: Why not split the success and failure cases into two lists rather than using the keys here? Using hard-coded strings for control flow is always tricky...

Comment (andrewyatz) [inline tests/unit/test_backends.py]: Fair point will refactor

**Comment by jeromekelleher** [comment]
+1. Merging.

Thanks @andrewyatz, looking forward to more PRs!
